test/runtests.jl

```diff
@@ -60,6 +60,16 @@ test_Ring_interface(S)
 	@test isone((S(1)//S(2))*2)
 
 	@test conj(a//c) == conj(a)//conj(c)
+
+	x=deepcopy(b)
+	ab=a+b
+	@test add!(x, a, x) == ab
+	@test x == ab
+
+	add!(a, a, a)
+	@test a == 2*b
+	add!(a, c, b)
+	@test a == c+b
```

Comment:
Ideally these tests should also verify that the other arguments are *not* changed.

Reply:
On the long rung we should just extend the ring interface tests to cover this, as we want this kind of test for *all* implementations... I.e. `test/Rings-conformance-tests.jl:193` in AA. Not asking you to do this right now, though

Reply:
(but see https://github.com/Nemocas/AbstractAlgebra.jl/issues/1814)

Reply:
Ok, I've added some extra tests for `add!` at least.